### ISSUE TITLE
Uncaught Error: cannot call methods on dialog prior to initialization; attempted to call method 'close'

### ISSUE THREAD

**Issue by liviuconcioiu** [phpmyadmin/phpmyadmin#17176]
### Describe the bug

Uncaught Error: cannot call methods on dialog prior to initialization; attempted to call method 'close' is received when trying to add an index and Preview SQL doesn't work.

### To Reproduce

1. Go to [https://demo.phpmyadmin.net/master-config/index.php?route=/table/structure&db=sakila&table=actor](https://demo.phpmyadmin.net/master-config/index.php?route=/table/structure&db=sakila&table=actor)
2. Click on Go button to create an index
3. Enter index name and click on Preview SQL (also preview sql doesn't work)
4. Click on Go
5. See error

### Expected behavior

Preview SQL should work and the error shouldn't be received.

### Screenshots

![index](https://user-images.githubusercontent.com/25424343/139758277-da8a4e8d-6272-4d89-82a8-8267afbff660.png)
From #17293

https://user-images.githubusercontent.com/25424343/149618180-6b8b5b7f-9a4d-4714-a3da-2ca860a1f8f3.mp4

### Server configuration

- Operating system:
- Web server:
- Database version:
- PHP version:
- phpMyAdmin version:

### Client configuration

- Browser: Firefox
- Operating system: Windows

### Additional context

```
{
    "pma_version": "5.2.0-dev",
    "browser_name": "FIREFOX",
    "browser_version": "95.0",
    "user_os": "Win",
    "server_software": "nginx/1.14.2",
    "user_agent_string": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:95.0) Gecko/20100101 Firefox/95.0",
    "locale": "en",
    "configuration_storage": "disabled",
    "php_version": "7.4.23",
    "script_name": "index.php",
    "exception_type": "js",
    "exception": {
        "mode": "stack",
        "name": "Error",
        "message": "cannot call methods on dialog prior to initialization; attempted to call method 'close'",
        "stack": [
            {
                "func": "error",
                "line": "2",
                "column": "2616",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "t.widget.bridge/t.fn[e]/<",
                "line": "6",
                "column": "5615",
                "context": [
                    "/*! jQuery UI - v1.12.1 - 2016-09-14",
                    "* http://jqueryui.com",
                    "* Includes: widget.js, position.js, data.js, disable-selection.js, effect.j//...",
                    "* Copyright jQuery Foundation and other contributors; Licensed MIT */",
                    "",
                    "(function(t){\"function\"==typeof define&&define.amd?define([\"jquery\"],t):t(j//...",
                    "}}(t.fn.show),hide:function(t){return function(s){if(i(s))return t.apply(th//...",
                    "this.isMultiLine=o||!a&&this._isContentEditable(this.element),this.valueMet//...",
                    "}},_updateDatepicker:function(e){this.maxRows=4,m=e,e.dpDiv.empty().append(//...",
                    "},_cacheMargins:function(){this.margins={left:parseInt(this.element.css(\"ma//...",
                    "},disable:t.noop,enable:t.noop,close:function(e){var i=this;this._isOpen&&t//..."
                ],
                "uri": "js/vendor/jquery/jquery-ui.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery-ui.min.js"
            },
            {
                "func": "each",
                "line": "2",
                "column": "3003",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "each",
                "line": "2",
                "column": "1481",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "t.widget.bridge/t.fn[e]",
                "line": "6",
                "column": "5357",
                "context": [
                    "/*! jQuery UI - v1.12.1 - 2016-09-14",
                    "* http://jqueryui.com",
                    "* Includes: widget.js, position.js, data.js, disable-selection.js, effect.j//...",
                    "* Copyright jQuery Foundation and other contributors; Licensed MIT */",
                    "",
                    "(function(t){\"function\"==typeof define&&define.amd?define([\"jquery\"],t):t(j//...",
                    "}}(t.fn.show),hide:function(t){return function(s){if(i(s))return t.apply(th//...",
                    "this.isMultiLine=o||!a&&this._isContentEditable(this.element),this.valueMet//...",
                    "}},_updateDatepicker:function(e){this.maxRows=4,m=e,e.dpDiv.empty().append(//...",
                    "},_cacheMargins:function(){this.margins={left:parseInt(this.element.css(\"ma//...",
                    "},disable:t.noop,enable:t.noop,close:function(e){var i=this;this._isOpen&&t//..."
                ],
                "uri": "js/vendor/jquery/jquery-ui.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery-ui.min.js"
            },
            {
                "func": "Functions.indexDialogModal/</<",
                "line": "3482",
                "column": "28",
                "context": [
                    "        $('#table_index').remove();",
                    "        $('<div id=\\'temp_div\\'><div>').append(data.index_table).find('#tab//...",
                    "        var $editIndexDialog = $('#indexDialogModal');",
                    "",
                    "        if ($editIndexDialog.length > 0) {",
                    "          $editIndexDialog.dialog('close');",
                    "        }",
                    "",
                    "        $('div.no_indexes_defined').hide();",
                    "",
                    "        if (callbackSuccess) {"
                ],
                "uri": "js/dist/functions.js?v=5.2.0-dev",
                "scriptname": "js/dist/functions.js"
            },
            {
                "func": "c",
                "line": "2",
                "column": "28327",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "fireWith",
                "line": "2",
                "column": "29072",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "l",
                "line": "2",
                "column": "79901",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "send/o/<",
                "line": "2",
                "column": "82355",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "EventHandlerNonNull*send",
                "line": "2",
                "column": "82541",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "ajax",
                "line": "2",
                "column": "78291",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "jQuery.ajax",
                "line": "305",
                "column": "22",
                "context": [
                    "",
                    "var oldAjax = jQuery.ajax,",
                    "\trjsonp = /(=)\\?(?=&|$)|\\?\\?/;",
                    "",
                    "jQuery.ajax = function( ) {",
                    "\tvar jQXHR = oldAjax.apply( this, arguments );",
                    "",
                    "\t// Be sure we got a jQXHR (e.g., not sync)",
                    "\tif ( jQXHR.promise ) {",
                    "\t\tmigrateWarnFunc( jQXHR, \"success\", jQXHR.done,",
                    "\t\t\t\"jQXHR.success is deprecated and removed\" );"
                ],
                "uri": "js/vendor/jquery/jquery-migrate.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery-migrate.js"
            },
            {
                "func": "S[i]",
                "line": "2",
                "column": "80344",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "Functions.indexDialogModal/<",
                "line": "3464",
                "column": "7",
                "context": [
                    "     */",
                    "    var $form = $('#index_frm');",
                    "    Functions.ajaxShowMessage(Messages.strProcessingRequest);",
                    "    Functions.prepareForAjaxRequest($form); // User wants to submit the form",
                    "",
                    "    $.post($form.attr('action'), $form.serialize() + CommonParams.get('arg_//...",
                    "      var $sqlqueryresults = $('.sqlqueryresults');",
                    "",
                    "      if ($sqlqueryresults.length !== 0) {",
                    "        $sqlqueryresults.remove();",
                    "      }"
                ],
                "uri": "js/dist/functions.js?v=5.2.0-dev",
                "scriptname": "js/dist/functions.js"
            },
            {
                "func": "newFunc",
                "line": "258",
                "column": "23",
                "context": [
                    "   */",
                    "  wrapFunction: function (func) {",
                    "    if (!func.wrapped) {",
                    "      var newFunc = function () {",
                    "        try {",
                    "          return func.apply(this, arguments);",
                    "        } catch (x) {",
                    "          TraceKit.report(x);",
                    "        }",
                    "      };",
                    ""
                ],
                "uri": "js/dist/error_report.js?v=5.2.0-dev",
                "scriptname": "js/dist/error_report.js"
            },
            {
                "func": "dispatch",
                "line": "2",
                "column": "43064",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "add/v.handle",
                "line": "2",
                "column": "41048",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "EventListener.handleEvent*add",
                "line": "2",
                "column": "41515",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "jQuery.event.add",
                "line": "642",
                "column": "21",
                "context": [
                    "",
                    "\t// This misses the multiple-types case but that seems awfully rare",
                    "\tif ( elem === window && types === \"load\" && window.document.readyState ===//...",
                    "\t\tmigrateWarn( \"jQuery(window).on('load'...) called after load event occurr//...",
                    "\t}",
                    "\treturn oldEventAdd.apply( this, arguments );",
                    "};",
                    "",
                    "jQuery.each( [ \"load\", \"unload\", \"error\" ], function( _, name ) {",
                    "",
                    "\tjQuery.fn[ name ] = function() {"
                ],
                "uri": "js/vendor/jquery/jquery-migrate.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery-migrate.js"
            },
            {
                "func": "Ee/<",
                "line": "2",
                "column": "40109",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "each",
                "line": "2",
                "column": "3003",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "each",
                "line": "2",
                "column": "1481",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "Ee",
                "line": "2",
                "column": "40085",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "on",
                "line": "2",
                "column": "46578",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "wrapJqueryOnCallback/$.fn.on",
                "line": "304",
                "column": "20",
                "context": [
                    "          arguments[i] = ErrorReport.wrapFunction(arguments[i]);",
                    "          break;",
                    "        }",
                    "      }",
                    "",
                    "      return oldOn.apply(this, arguments);",
                    "    };",
                    "  },",
                    "",
                    "  /**",
                    "   * Wraps the callback in AJAX.registerOnload automatically"
                ],
                "uri": "js/dist/error_report.js?v=5.2.0-dev",
                "scriptname": "js/dist/error_report.js"
            },
            {
                "func": "wrapJqueryOnCallback/$.fn.on",
                "line": "304",
                "column": "20",
                "context": [
                    "          arguments[i] = ErrorReport.wrapFunction(arguments[i]);",
                    "          break;",
                    "        }",
                    "      }",
                    "",
                    "      return oldOn.apply(this, arguments);",
                    "    };",
                    "  },",
                    "",
                    "  /**",
                    "   * Wraps the callback in AJAX.registerOnload automatically"
                ],
                "uri": "js/dist/error_report.js?v=5.2.0-dev",
                "scriptname": "js/dist/error_report.js"
            },
            {
                "func": "wrapJqueryOnCallback/$.fn.on",
                "line": "304",
                "column": "20",
                "context": [
                    "          arguments[i] = ErrorReport.wrapFunction(arguments[i]);",
                    "          break;",
                    "        }",
                    "      }",
                    "",
                    "      return oldOn.apply(this, arguments);",
                    "    };",
                    "  },",
                    "",
                    "  /**",
                    "   * Wraps the callback in AJAX.registerOnload automatically"
                ],
                "uri": "js/dist/error_report.js?v=5.2.0-dev",
                "scriptname": "js/dist/error_report.js"
            },
            {
                "func": "wrapJqueryOnCallback/$.fn.on",
                "line": "304",
                "column": "20",
                "context": [
                    "          arguments[i] = ErrorReport.wrapFunction(arguments[i]);",
                    "          break;",
                    "        }",
                    "      }",
                    "",
                    "      return oldOn.apply(this, arguments);",
                    "    };",
                    "  },",
                    "",
                    "  /**",
                    "   * Wraps the callback in AJAX.registerOnload automatically"
                ],
                "uri": "js/dist/error_report.js?v=5.2.0-dev",
                "scriptname": "js/dist/error_report.js"
            },
            {
                "func": "Functions.indexDialogModal",
                "line": "3456",
                "column": "34",
                "context": [
                    "  /**",
                    "   * @var button_options Object that stores the options",
                    "   *                     passed to jQueryUI dialog",
                    "   */",
                    "",
                    "  $('#indexDialogModalGoButton').on('click', function () {",
                    "    /**",
                    "     * @var the_form object referring to the export form",
                    "     */",
                    "    var $form = $('#index_frm');",
                    "    Functions.ajaxShowMessage(Messages.strProcessingRequest);"
                ],
                "uri": "js/dist/functions.js?v=5.2.0-dev",
                "scriptname": "js/dist/functions.js"
            },
            {
                "func": "Functions.indexEditorDialog",
                "line": "3534",
                "column": "13",
                "context": [
                    "    }",
                    "  }); // end $.get()",
                    "};",
                    "",
                    "Functions.indexEditorDialog = function (url, title, callbackSuccess, callba//...",
                    "  Functions.indexDialogModal('index.php?route=/table/indexes', url, title, //...",
                    "};",
                    "",
                    "Functions.indexRenameDialog = function (url, title, callbackSuccess, callba//...",
                    "  Functions.indexDialogModal('index.php?route=/table/indexes/rename', url, //...",
                    "};"
                ],
                "uri": "js/dist/functions.js?v=5.2.0-dev",
                "scriptname": "js/dist/functions.js"
            },
            {
                "func": "?",
                "line": "691",
                "column": "15",
                "context": [
                    "      url = $(this).find('a').getPostData();",
                    "      title = Messages.strEditIndex;",
                    "    }",
                    "",
                    "    url += CommonParams.get('arg_separator') + 'ajax_request=true';",
                    "    Functions.indexEditorDialog(url, title, function (data) {",
                    "      CommonParams.set('db', data.params.db);",
                    "      CommonParams.set('table', data.params.table);",
                    "      CommonActions.refreshMain('index.php?route=/table/structure');",
                    "    });",
                    "  });"
                ],
                "uri": "js/dist/indexes.js?v=5.2.0-dev",
                "scriptname": "js/dist/indexes.js"
            },
            {
                "func": "newFunc",
                "line": "258",
                "column": "23",
                "context": [
                    "   */",
                    "  wrapFunction: function (func) {",
                    "    if (!func.wrapped) {",
                    "      var newFunc = function () {",
                    "        try {",
                    "          return func.apply(this, arguments);",
                    "        } catch (x) {",
                    "          TraceKit.report(x);",
                    "        }",
                    "      };",
                    ""
                ],
                "uri": "js/dist/error_report.js?v=5.2.0-dev",
                "scriptname": "js/dist/error_report.js"
            },
            {
                "func": "dispatch",
                "line": "2",
                "column": "43064",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "add/v.handle",
                "line": "2",
                "column": "41048",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.2.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            }
        ],
        "uri": "index.php?route=%2Ftable%2Fstructure"
    }
}
```

**Comment by williamdes** [comment]
On 5.2 I can not confirm any bug, do you ?
Can you try on 5.3 ?

**Comment by liviuconcioiu** [comment]
@williamdes it seems to be fixed on both 5.2 and 5.3.
